File: ios/scanmynet_sdk/Sources/scanmynet_sdk/ScanmynetSdkPlugin.swift
import Flutter
import UIKit

/// ScanMyNet plugin entry point.
///
/// Registers the Pigeon-generated `ScanHostApi` handler (Dart -> native) and
/// constructs the `ScanFlutterApi` caller (native -> Dart) used to push scan
/// events back to Flutter.
public class ScanmynetSdkPlugin: NSObject, FlutterPlugin {
  private var hostApi: ScanHostApiImpl?

  public static func register(with registrar: FlutterPluginRegistrar) {
    let messenger = registrar.messenger()
    let instance = ScanmynetSdkPlugin()
    let flutterApi = ScanFlutterApi(binaryMessenger: messenger)
    instance.hostApi = ScanHostApiImpl(flutterApi: flutterApi)
    ScanHostApiSetup.setUp(binaryMessenger: messenger, api: instance.hostApi)
    // Retain the plugin (and its host-api impl) for the engine's lifetime.
    registrar.publish(instance)
  }
}

File: android/src/main/kotlin/com/creativeadvtech/scanmynet_sdk/ScanmynetSdkPlugin.kt
package com.creativeadvtech.scanmynet_sdk

import io.flutter.embedding.engine.plugins.FlutterPlugin

/**
 * ScanMyNet plugin entry point.
 *
 * Registers the Pigeon-generated [ScanHostApi] handler (Dart -> native) and
 * constructs the [ScanFlutterApi] caller (native -> Dart) used to push scan
 * events back to Flutter.
 */
class ScanmynetSdkPlugin : FlutterPlugin {
    private var hostApi: ScanHostApiImpl? = null

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        val flutterApi = ScanFlutterApi(binding.binaryMessenger)
        hostApi = ScanHostApiImpl(binding.applicationContext, flutterApi)
        ScanHostApi.setUp(binding.binaryMessenger, hostApi)
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        ScanHostApi.setUp(binding.binaryMessenger, null)
        hostApi?.dispose()
        hostApi = null
    }
}
